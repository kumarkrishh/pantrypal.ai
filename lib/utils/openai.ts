'use client';

export const createOpenAIClient = () => null; // Stubbed for compatibility

export const processImageForIngredients = async (
  _unused: any,
  base64Image: string
): Promise<string[]> => {
  console.log('[processImageForIngredients] Starting image analysis...');

  try {
    const response = await fetch('/api/detectIngredients', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ base64Image }),
    });

    console.log('[processImageForIngredients] Response status:', response.status);

    const json = await response.json();
    console.log('[processImageForIngredients] Response JSON:', json);

    if (!response.ok) {
      throw new Error(json.error || 'Image processing failed');
    }

    const { ingredients } = json;
    console.log('[processImageForIngredients] Extracted ingredients:', ingredients);

    return ingredients;
  } catch (err) {
    console.error('[processImageForIngredients] Error:', err);
    throw err;
  }
};
