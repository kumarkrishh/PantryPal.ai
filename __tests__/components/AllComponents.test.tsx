import { testComponentRendering } from '../utils/testUtils';
import LoginForm from '@/components/LoginForm';
// Import all your components here

// Mock common hooks and services
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    // Add other router methods you use
  })
}));

// Add other global mocks here

// Test each component with various prop combinations
testComponentRendering(LoginForm, [
  {},
  { onSuccess: () => {} },
  { onError: () => {} },
  { isLoading: true },
]);

// Add more components with their prop combinations 