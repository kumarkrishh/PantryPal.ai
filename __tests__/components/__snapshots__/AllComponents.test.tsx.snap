// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm - Basic Rendering matches snapshot 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full max-w-md border-indigo-100"
  >
    <div
      class="flex flex-col p-6 space-y-1"
    >
      <h3
        class="tracking-tight text-2xl font-bold bg-gradient-to-r from-indigo-600 to-purple-600 bg-clip-text text-transparent"
      >
        Welcome back
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        Choose your preferred sign in method
      </p>
    </div>
    <div
      class="p-6 pt-0 space-y-4"
    >
      <div
        class="grid grid-cols-1 gap-3"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border bg-background h-10 px-4 py-2 w-full border-indigo-100 hover:bg-indigo-50 hover:text-indigo-600 hover:border-indigo-200"
        >
          <svg
            aria-hidden="true"
            class="mr-2 h-4 w-4"
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 48 48"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
              fill="#FFC107"
            />
            <path
              d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
              fill="#FF3D00"
            />
            <path
              d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
              fill="#4CAF50"
            />
            <path
              d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
              fill="#1976D2"
            />
          </svg>
          Continue with Google
        </button>
      </div>
      <div
        class="relative"
      >
        <div
          class="absolute inset-0 flex items-center"
        >
          <span
            class="w-full border-t border-indigo-100"
          />
        </div>
        <div
          class="relative flex justify-center text-xs uppercase"
        >
          <span
            class="bg-background px-2 text-muted-foreground"
          >
            Or continue with
          </span>
        </div>
      </div>
      <form
        class="space-y-4"
      >
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for="email"
          >
            Email
          </label>
          <input
            class="flex h-10 w-full rounded-md border bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border-indigo-100 focus-visible:ring-indigo-600"
            id="email"
            name="email"
            placeholder="m@example.com"
            required=""
            type="email"
          />
        </div>
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for="password"
          >
            Password
          </label>
          <input
            class="flex h-10 w-full rounded-md border bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border-indigo-100 focus-visible:ring-indigo-600"
            id="password"
            name="password"
            required=""
            type="password"
          />
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary hover:bg-primary/90 h-10 px-4 py-2 w-full bg-gradient-to-r from-indigo-600 to-purple-600 hover:from-indigo-700 hover:to-purple-700 text-white"
          type="submit"
        >
          Sign in
          <svg
            class="lucide lucide-arrow-right ml-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="m12 5 7 7-7 7"
            />
          </svg>
        </button>
      </form>
    </div>
    <div
      class="p-6 pt-0 flex flex-wrap items-center justify-between gap-2"
    >
      <div
        class="text-sm text-muted-foreground"
      >
        <span
          class="mr-1"
        >
          Don't have an account?
        </span>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 underline-offset-4 hover:underline h-10 p-0 text-indigo-600 hover:text-indigo-700"
        >
          Sign up
        </button>
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 underline-offset-4 hover:underline h-10 text-sm text-muted-foreground hover:text-indigo-600 p-0"
      >
        Forgot password?
      </button>
    </div>
  </div>
</div>
`;
